libmamba/src/api/install.cpp

```diff
@@ -715,5 +715,23 @@ namespace mamba
                 }
             }
         }
+
+        void channels_hook(std::vector<std::string>& channels)
+        {
+            auto& config = Configuration::instance();
+
+            if (find(channels.begin(), channels.end(), "nodefaults") != channels.end())
+            {
+                if (config.at("channels").cli_configured())
+                {
+                    channels = config.at("channels").cli_value<std::vector<std::string>>();
+                    channels.erase(find(channels.begin(), channels.end(), "nodefaults"));
+                }
+                else
+                {
+                    channels.clear();
```

Comment:
Is that correct, to clear the entire `channels` vector?

Reply:
Clearing the `channels` vector is what the previous `override_channels` hook did, so it should still be correct.
`nodefaults` means only channels passed explicitly (cli and environment.yml) should be used. If no channels are passed explicitly and we won't use non-explicitly passed channels then we don't have any channels.


`override_channels` isn't used anymore, as the `override_channels` hook now appends `nodefaults` to the channels. `--override-channels` on the cli and `nodefaults` in environment.yml` do the same thing.